src/core/control/settings/SettingsDescription.h

```diff
@@ -0,0 +1,1794 @@
+    using getter_return_type = bool;
+    static constexpr auto xmlName = "pressureGuessing";
+    static constexpr value_type DEFAULT = false;
+    static constexpr auto COMMENT = nullptr;
+    static constexpr auto IMPORT_FN = importBoolProperty;
+    static constexpr auto EXPORT_FN = exportBoolProperty;
+    static constexpr auto VALIDATE_FN = noValidate<value_type>;
+};
+
+template <>
+struct Setting<SettingsElement::SETTING_SCROLLBAR_HIDE_TYPE> {
+    using value_type = ScrollbarHideType;
+    using getter_return_type = ScrollbarHideType;
+    static constexpr auto xmlName = "scrollbarHideType";
+    static constexpr value_type DEFAULT = ScrollbarHideType::SCROLLBAR_HIDE_NONE;
+    static constexpr auto COMMENT =
+            "Hides scroolbars in the main window, allowed values: \"none\", \"horizontal\", \"vertical\", \"both\"";
+    static constexpr auto IMPORT_FN = importScrollbarHideType;
+    static constexpr auto EXPORT_FN = exportScrollbarHideType;
+    static constexpr auto VALIDATE_FN = noValidate<value_type>;
+};
+
+template <>
+struct Setting<SettingsElement::SETTING_DISABLE_SCROLLBAR_FADEOUT> {
+    using value_type = bool;
+    using getter_return_type = bool;
+    static constexpr auto xmlName = "disableScrollbarFadeout";
+    static constexpr value_type DEFAULT = false;
+    static constexpr auto COMMENT = nullptr;
+    static constexpr auto IMPORT_FN = importBoolProperty;
+    static constexpr auto EXPORT_FN = exportBoolProperty;
+    static constexpr auto VALIDATE_FN = noValidate<value_type>;
+};
+
+template <>
+struct Setting<SettingsElement::SETTING_DISABLE_AUDIO> {
+    using value_type = bool;
+    using getter_return_type = bool;
+    static constexpr auto xmlName = "disableAudio";
+    static constexpr value_type DEFAULT = false;
+    static constexpr auto COMMENT = nullptr;
+    static constexpr auto IMPORT_FN = importBoolProperty;
+    static constexpr auto EXPORT_FN = exportBoolProperty;
+    static constexpr auto VALIDATE_FN = noValidate<value_type>;
+};
+
+template <>
+struct Setting<SettingsElement::SETTING_AUDIO_SAMPLE_RATE> {
+    using value_type = double;
+    using getter_return_type = double;
+    static constexpr auto xmlName = "audioSampleRate";
+    static constexpr value_type DEFAULT = 44100.0;
+    static constexpr auto COMMENT = nullptr;
+    static constexpr auto IMPORT_FN = importDoubleProperty;
+    static constexpr auto EXPORT_FN = exportDoubleProperty;
+    static constexpr auto VALIDATE_FN = noValidate<value_type>;
+};
+
+template <>
+struct Setting<SettingsElement::SETTING_AUDIO_GAIN> {
+    using value_type = double;
+    using getter_return_type = double;
+    static constexpr auto xmlName = "audioGain";
+    static constexpr value_type DEFAULT = 1.0;
+    static constexpr auto COMMENT = nullptr;
+    static constexpr auto IMPORT_FN = importDoubleProperty;
+    static constexpr auto EXPORT_FN = exportDoubleProperty;
+    static constexpr auto VALIDATE_FN = noValidate<value_type>;
+};
```

Comment:
I implemented all three of these, but I thaught about maybe using the getter return type as the type given to the export functions, what would you think about doing it like this instead of always using a const&?

Reply:
> I implemented all three of these, but I thaught about maybe using the getter return type as the type given to the export functions, what would you think about doing it like this instead of always using a const&?

I thought more about this and implemented it, as I think it is the better choice.